### PR TITLE
Add a new factory function :  make_fixed_width_column()

### PR DESCRIPTION
**Pull request by nvdbaranec**
Adds an overload to allocate_like() that takes raw inputs instead of deriving from an input column. The primary use case for this is callers of the form:
```
out column = caller(scalar, scalar, column);
```
where the type of the output column is derived from the scalar inputs, but the size comes from the column input.  Concrete example:
```
copy_if_else(scalar lhs, scalar rhs, column boolean_mask)
{
   out = allocate_like(lhs.type(), mask.size());
}
```

### CODE REVIEW

**Comment by jrhemstad** [comment]
Instead of an `allocate_like`, this really should be a new factory:
```c++
unique_ptr<column> make_fixed_width_column(data_type type, size_type size, mask_state state, ...);
```

I originally split `make_numeric_column` and `make_timestamp_column` because I wasn't sure if the timestamp column would require additional arguments. But now that I've had some time to think about it, I'm pretty sure any additional arguments would be encapsulated into the `data_type` argument.

As such, I'd just make this code into the new factory.

File: cpp/src/column/column_factories.cpp [hunk 1]
```diff
@@ -64,4 +64,16 @@ std::unique_ptr<column> make_timestamp_column(
       std::vector<std::unique_ptr<column>>{});
 }
 
+// Allocate storage for a specified number of fixed width elements
+std::unique_ptr<column> make_fixed_width_column(
+    data_type type, size_type size, mask_state state, cudaStream_t stream,
+    rmm::mr::device_memory_resource* mr) {
+  CUDF_EXPECTS(is_fixed_width(type), "Invalid, non-fixed-width type.");
+
+  return std::make_unique<column>(
```
Comment (jrhemstad) [inline cpp/src/column/column_factories.cpp]: This should invoke either `make_numeric_column` or `make_timestamp_column`.

File: cpp/include/cudf/column/column_factories.hpp [hunk 2]
```diff
@@ -138,6 +138,64 @@ std::unique_ptr<column> make_timestamp_column(
...
+template <typename B>
+std::unique_ptr<column> make_fixed_width_column(
+    data_type type, size_type size,
+    B&& null_mask,
+    size_type null_count = cudf::UNKNOWN_NULL_COUNT, cudaStream_t stream = 0,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource())
+{
+  CUDF_EXPECTS(is_fixed_width(type), "Invalid, non-fixed-width type.");
+  if(is_timestamp(type)){
+    return make_timestamp_column(type, size, null_mask, null_count, stream, mr);
```
Comment (jrhemstad) [inline cpp/include/cudf/column/column_factories.hpp]:
```suggestion
    return make_timestamp_column(type, size, std::forward<B>(null_mask), null_count, stream, mr);
```

File: cpp/include/cudf/column/column_factories.hpp [hunk 3]
```diff
@@ -138,6 +138,64 @@ std::unique_ptr<column> make_timestamp_column(
...
+  }
+  return make_numeric_column(type, size, null_mask, null_count, stream, mr);
```
Comment (jrhemstad) [inline cpp/include/cudf/column/column_factories.hpp]:
```suggestion
  return make_numeric_column(type, size, std::forward<B>(null_mask), null_count, stream, mr);  
```

File: cpp/include/cudf/detail/copy.hpp [hunk 4]
```diff
@@ -90,6 +90,23 @@ std::unique_ptr<column> allocate_like(column_view const& input, size_type size,
...
+ * Supports only fixed-width types.
+ *
+ * @param[in] type The type of elements in the new column
+ * @param[in] size The desired number of elements that the new column should have capacity for
+ * @param[in] nullable Optional, whether or not the output column should contain a null mask
+ * @param[in] mr Optional, The resource to use for all allocations
+ * @param[in] stream Optional CUDA stream on which to execute kernels
+ * @return std::unique_ptr<column> A column with sufficient uninitialized capacity to hold the specified number of elements as `input` of the same type as `input.type()`
+ */
+std::unique_ptr<column> allocate_like(data_type type, size_type size,
```
Comment (jrhemstad) [inline cpp/include/cudf/detail/copy.hpp]: This shouldn't need to exist anymore?

Comment (nvdbaranec) [reply cpp/include/cudf/detail/copy.hpp]: Yep, sorry.  Missed that one.

**Comment by harrism** [comment]
Just realized this is in branch-0.12. Is it needed in branch-0.11?

**Pull request merged**